assets/js/datasets.js

```diff
@@ -0,0 +1,186 @@
+'use strict';
+
+window.linqs = window.linqs || {};
+window.linqs.utils = window.linqs.utils || {};
+window.linqs.datasets = window.linqs.datasets || {};
+window.linqs.datasets.metadata = window.linqs.datasets.metadata || {};
+window.linqs.datasets.pubs = window.linqs.datasets.pubs || {};
+
+window.linqs.datasets.makeLink = function(path) {
+	if (path.startsWith('/')) {
+		return window.linqs.datasets.baseURL.replace(/\/$/, '') + path;
+	}
+
+	return path;
+};
+
+window.linqs.datasets.listAuthors = function(pub) {
+	let authors = '';
+
+	for (let i = 0; i < pub['authors'].length; i++) {
+		let text = pub['authors'][i];
+
+		if (i == 0) {
+			authors += text;
+		} else {
+			authors += ' and ' + text;
+		}
+	}
+
+	return authors;
+};
+
+window.linqs.datasets.bibtex = function(pub, id) {
+	let authors = window.linqs.datasets.listAuthors(pub);
+	return window.linqs.utils.bibtex(pub, authors, id);
+};
+
+window.linqs.datasets.size = function(size) {
+	let sizes = ["B", "KB", "MB", "GB", "TB", "PB"];
+
+	let i = 0;
+	while (i < sizes.length && size >= 1024) {
+		size = size / 1024.0;
+		i++;
+	}
+
+	return size.toFixed(2).toString().replace(/0+$/, '') + " " +sizes[i];
+};
+
+window.linqs.datasets.makeDownloadInfo = function(downloads) {
+	let downloadInfosHTML = '';
+
+	downloads.forEach(function(download, i) {
+		let link = download['href'];
+		let text = download["text"];
+		let statsHTML = "";
+
+		let sizeRaw = download['size'];
+		let md5 = download['md5'];
+
+		if (sizeRaw != null && md5 != null) {
+			let size = window.linqs.datasets.size(sizeRaw);
+
+			statsHTML = `
+				<div class='size'>
+					(${size})
+				</div>
+				<div class='hash'>
+					md5: ${md5}
+				</div>
+			`;
+		}
+
+		let downloadInfoHTML = `
+			<div class='link-info'>
+				<a class='download-link' href='${link}'>${text}</a>
+				${statsHTML}
+			</div>
+		`;
+
+		downloadInfosHTML += downloadInfoHTML;
+	});
+
+	return downloadInfosHTML;
+};
+
+window.linqs.datasets.makeReferences = function(references) {
+	let referencesHTML = '';
+
+	references.forEach(function(reference, i) {
+		let refLink = window.linqs.datasets.makeLink(reference['href']);
+		let refText = reference['text'];
+		let referenceTemplate = `
+			<p><a href='${refLink}'>${refText}</a></p>
```

Comment:
Why `<p>`?
Seems unnecessary.

Reply:
Without the `<p>` all the references are just stuck together on the same line.

Reply:
Then we can use css to properly style it.
Remember, data in html and style in css.

Using the right tags is important for a lot of reasons (like accessibility and screen readers).
You don't want to use tags where they semantically don't make sense.